cmd/agent/app/reporter/grpc/reporter.go

```diff
@@ -76,20 +76,15 @@ func (r *Reporter) send(spans []*model.Span, process *model.Process) error {
 	return err
 }
 
-// addTags appends jaeger tags for the agent to every span it sends to the collector.
-func addProcessTags(spans []*model.Span, process *model.Process, agentTags []model.KeyValue) ([]*model.Span, *model.Process) {
-	if len(agentTags) == 0 {
-		return spans, process
-	}
-	if process != nil {
-		process.Tags = append(process.Tags, agentTags...)
-	}
-	for _, span := range spans {
-		if span.Process != nil {
-			span.Process.Tags = append(span.Process.Tags, agentTags...)
+func checkIfPresentAlready(tags []model.KeyValue, agentTag model.KeyValue) (int, bool) {
+	i := 0
+	for _, tag := range tags {
+		if tag.Key == agentTag.Key {
+			return i, true
 		}
+		i++
 	}
-	return spans, process
+	return -1, false
```

Comment:
`return 0, false`: the first arg does not need an extra indication (via -1) that tag was not found, it's what the boolean is for